include/jemalloc/internal/extent.h

```diff
@@ -188,9 +188,15 @@ struct extent_s {
 		/* Profiling data, used for large objects. */
 		struct {
 			/* Time when this was allocated. */
-			nstime_t		e_alloc_time;
+			nstime_t	e_alloc_time;
 			/* Points to a prof_tctx_t. */
-			atomic_p_t		e_prof_tctx;
+			atomic_p_t	e_prof_tctx;
+			/*
+			 * Points to a prof_recent_t for the allocation; NULL
+			 * means the recent allocation record no longer exists.
+			 * Protected by prof_recent_alloc_mtx.
+			 */
+			prof_recent_t	*e_prof_recent_alloc;
```

Comment:
I think you meant `sizeof(slab_data_t) >= sizeof("prof_struct")`? I added the assert in `extent_init()` (by "abusing" `cassert()` for now).

You're right that we didn't name the struct, which most likely, I believe, is because that we could save one indirection. But I don't think this matters much because only the getters and setters are slightly affected.